File: app/src/main/java/com/example/bskl_kotlin/fragment/attendance/adapter/StrudentSpinnerAdapter.java
package com.example.bskl_kotlin.fragment.attendance.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.bskl_kotlin.R;
import com.example.bskl_kotlin.fragment.absence.model.StudentModel;
import com.example.bskl_kotlin.fragment.attendance.model.StudentModelJava;
import com.example.bskl_kotlin.manager.AppUtils;

import java.util.ArrayList;

public class StrudentSpinnerAdapter extends RecyclerView.Adapter<StrudentSpinnerAdapter.MyViewHolder> {

    private Context mContext;
    private ArrayList<StudentModelJava> mStudentList;
    String dept;


    public class MyViewHolder extends RecyclerView.ViewHolder {
        TextView mTitleTxt,listTxtClass;
        ImageView imgIcon;
        RelativeLayout relSub;

        public MyViewHolder(View view) {
            super(view);

            mTitleTxt= view.findViewById(R.id.listTxtTitle);
            listTxtClass= view.findViewById(R.id.listTxtClass);
            imgIcon= view.findViewById(R.id.imagicon);
            relSub= view.findViewById(R.id.relSub);
        }
    }


    public StrudentSpinnerAdapter(Context mContext, ArrayList<StudentModelJava> mStudentList) {
        this.mContext = mContext;
        this.mStudentList = mStudentList;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.fragment_student_list_adapter, parent, false);

        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(final MyViewHolder holder, int position) {
        if (mStudentList.get(position).getAlumi().equalsIgnoreCase("0")) {
            holder.relSub.setVisibility(View.VISIBLE);
            holder.mTitleTxt.setText(mStudentList.get(position).getName());
            holder.imgIcon.setVisibility(View.VISIBLE);
            holder.listTxtClass.setText(mStudentList.get(position).getmClass());
            if (!mStudentList.get(position).getPhoto().equals("")) {

                Glide.with(mContext).load(mStudentList.get(position).getPhoto().toString()).placeholder(R.drawable.boy).into(holder.imgIcon);
            } else

            {

                holder.imgIcon.setImageResource(R.drawable.boy);
            }
        }
        else
        {
            holder.relSub.setVisibility(View.GONE);

        }
    }

    @Override
    public int getItemCount() {
        return mStudentList.size();
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }
}

File: app/src/main/java/com/example/bskl_kotlin/fragment/absence/adapter/StrudentSpinnerAdapter.kt
package com.example.bskl_kotlin.fragment.absence.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.bskl_kotlin.R
import com.example.bskl_kotlin.fragment.absence.model.StudentModel

class StrudentSpinnerAdapter(private val mContext: Context, mStudentList: ArrayList<StudentModel>) :
    RecyclerView.Adapter<StrudentSpinnerAdapter.MyViewHolder>() {
    private val mStudentList: ArrayList<StudentModel>
    var dept: String? = null

    inner class MyViewHolder(view: View) : ViewHolder(view) {
        var mTitleTxt: TextView
        var listTxtClass: TextView
        var imgIcon: ImageView
        var relSub: RelativeLayout

        init {
            mTitleTxt = view.findViewById(R.id.listTxtTitle)
            listTxtClass = view.findViewById(R.id.listTxtClass)
            imgIcon = view.findViewById(R.id.imagicon)
            relSub = view.findViewById(R.id.relSub)
        }
    }

    init {
        this.mStudentList = mStudentList
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView: View = LayoutInflater.from(parent.context)
            .inflate(R.layout.fragment_student_list_adapter, parent, false)
        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        if (mStudentList[position].alumi.equals("0")) {
            holder.relSub.visibility = View.VISIBLE
            holder.mTitleTxt.setText(mStudentList[position].mName)
            holder.imgIcon.visibility = View.VISIBLE
            holder.listTxtClass.setText(mStudentList[position].mClass)
            if (!mStudentList[position].mPhoto.equals("")) {
                /*Picasso.with(mContext)
                    .load(AppUtils.replace(mStudentList[position].mPhoto.toString()))
                    .placeholder(R.drawable.boy).fit().into(holder.imgIcon)*/
            } else {
                holder.imgIcon.setImageResource(R.drawable.boy)
            }
        } else {
            holder.relSub.visibility = View.GONE
        }
    }

    override fun getItemCount(): Int {
        return mStudentList.size
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }
}